### PR TITLE
fix(domain): return update owner url if whoisOwner.id is defined

### PR DESCRIPTION
**Pull request by cbourgois**
## return update owner url if whoisOwner.id is defined

ref: MFRWW-900

### Description of the Change

In some case, `whoisOwner` is a string with an error message.

### CODE REVIEW

File: client/app/domain/domain-tabs.controller.js [hunk 1]
```diff
@@ -131,10 +131,7 @@ angular.module('App').controller(
     }
 
     static getUpdateOwnerUrl(domain) {
-      if (
-        _.get(domain, 'whoisOwner', false)
-        && domain.whoisOwner !== 'pending'
-      ) {
+      if (_.has(domain, 'whoisOwner.id')) {
```
Comment (FredericEspiau) [inline client/app/domain/domain-tabs.controller.js]: 👍

File: client/app/domain/domain-tabs.controller.js [hunk 2]
```diff
@@ -131,10 +131,7 @@ angular.module('App').controller(
     }
 
     static getUpdateOwnerUrl(domain) {
-      if (
-        _.get(domain, 'whoisOwner', false)
-        && domain.whoisOwner !== 'pending'
-      ) {
+      if (_.isObject(domain.whoisOwner) && _.has(domain, 'whoisOwner.id')) {
         return `#/useraccount/contact/${domain.name}/${domain.whoisOwner.id}`;
```
Comment (FredericEspiau) [inline client/app/domain/domain-tabs.controller.js]: You should also check if `domain` has a `name`

File: client/app/domain/domain-tabs.controller.js [hunk 3]
```diff
@@ -131,10 +131,7 @@ angular.module('App').controller(
     }
 
     static getUpdateOwnerUrl(domain) {
-      if (
-        _.get(domain, 'whoisOwner', false)
-        && domain.whoisOwner !== 'pending'
-      ) {
+      if (_.isObject(domain.whoisOwner) && _.has(domain, 'whoisOwner.id')) {
```
Comment (FredericEspiau) [inline client/app/domain/domain-tabs.controller.js]: Isn't `has` enough in the end ?

File: client/app/domain/domain-tabs.controller.js [hunk 4]
```diff
@@ -131,7 +131,7 @@ angular.module('App').controller(
     }
 
     static getUpdateOwnerUrl(domain) {
-      if (_.has(domain, 'whoisOwner.id')) {
+      if (_.isObject(domain.whoisOwner) && _.has(domain, 'whoisOwner.id')) {
```
[the remark this answers is not in the document]
Comment (ganeshkumar1989) [reply client/app/domain/domain-tabs.controller.js]: Sorry for the confusion. This change has been reverted

File: client/app/domain/domain-tabs.controller.js (hunk 3 above)
Comment (ganeshkumar1989) [reply client/app/domain/domain-tabs.controller.js]: Sorry for the confusion. This change has been reverted

File: client/app/domain/domain-tabs.controller.js (hunk 2 above)
Comment (ganeshkumar1989) [reply client/app/domain/domain-tabs.controller.js]: This check has been added

File: client/app/domain/domain-tabs.controller.js [hunk 5]
```diff
@@ -130,10 +132,19 @@ angular.module('App').controller(
...
     }
 
-    static getUpdateOwnerUrl(domain) {
-      if (_.isObject(domain.whoisOwner) && _.has(domain, 'whoisOwner.id')) {
+    getUpdateOwnerUrl(domain) {
+      if (_.has(domain, 'name') && _.has(domain, 'whoisOwner.id')) {
         return `#/useraccount/contact/${domain.name}/${domain.whoisOwner.id}`;
       }
+      this.$timeout(() => {
+        // Timeout is required so that the tabs get rendered before the error is shown
```
Comment (FredericEspiau) [inline client/app/domain/domain-tabs.controller.js]: :+1: nice telling us !

Comment (ganeshkumar1989) [reply client/app/domain/domain-tabs.controller.js]: Thanks! :)

File: client/app/domain/domain-tabs.controller.js [hunk 6]
```diff
@@ -130,14 +132,22 @@ angular.module('App').controller(
...
+      } else {
+        ownerUrlInfo.error = this.$scope.tr('domain_dashboard_whois_error', [
+          (!_.has(domain, 'name')) ? this.$scope.tr('domain_tab_REDIRECTION_add_step4_server_cname_error') : domain.whoisOwner,
+        ]);
+      }
+      return ownerUrlInfo;
+    }
+
+    handleOwnerUrlError(errMsg) {
+      if (errMsg) {
```
Comment (FredericEspiau) [inline client/app/domain/domain-tabs.controller.js]: You should throw an error when a parameter is incorrect

Comment (ganeshkumar1989) [reply client/app/domain/domain-tabs.controller.js]: We are already handling the error (if present), by showing an alerter. Do you think we will have to throw it once again?

**Comment by jleveugle** [comment]
Hello @cbourgois,
Your pull request is now closed due to inactivity.
Feel free to re-open it if you have any update.
Anna Bot for OVH UX <3

**Comment by cbourgois** [comment]
@ganeshkumar1989 there is some conflicts

**Comment by ganeshkumar1989** [comment]
Conflicts are fixed.

**Comment by jleveugle** [comment]
Hello @cbourgois,
Your pull request is now closed due to inactivity.
Feel free to re-open it if you have any update.
Anna Bot for OVH UX <3

**Pull request merged**